### ISSUE TITLE
[BUG] PaymentSheet and SEPA noPaymentMethodTypesAvailable error 

### ISSUE THREAD

**Issue by irakliitavadze** [stripe/stripe-ios#2100]
## Summary
Getting the "noPaymentMethodTypesAvailable" error during paymentSheet presentation. As I understand there is only one option to implement SEPA pay later through PaymentSheet. Any ideas why it doesn't work and what should I do?

## Code to reproduce
        StripeAPI.defaultPublishableKey = viewModel.publicKey
        var configuration = PaymentSheet.Configuration()
        configuration.merchantDisplayName = "Example, Inc."
        configuration.allowsDelayedPaymentMethods = true
        configuration.defaultBillingDetails.address.country = "DE"
        configuration.defaultBillingDetails.email = "test@email.com"
        configuration.defaultBillingDetails.name = "TestName"
        configuration.defaultBillingDetails.phone = "+494242424242"
     
        guard let secret = viewModel.paymentInitiate?.pspSecretToken else { return }
        self.paymentSheet = PaymentSheet(setupIntentClientSecret: secret, configuration: configuration)
        
        paymentSheet?.present(from: self) { paymentResult in
            // MARK: Handle the payment result
            switch paymentResult {
            case .completed:
              print("Your order is confirmed")
            case .canceled:
              print("Canceled!")
            case .failed(let error):
              print("Payment failed: \(error)")
            }
          } 

## iOS version
15.0+

## Installation method
SPM

## SDK version
23.1.1

## Log
 LOG ANALYTICS: ["app_name": "Garantiemax", "analytics_ua": "analytics.stripeios-1.0", "publishable_key": "pk_test_51KPRR3HPa93bcSPTQ9U7kPyUbHnup2qRPv7sdYZSk67WKZ9rCyyxFv9VPmwZ8a4EF7zN276vPsBoiRNxBDWJ71xt00oBjn1YuL", "pay_var": "legacy", "install": "S", "event": "luxe_client_filtered_payment_methods", "product_usage": ["PaymentSheet"], "ocr_type": "none", "bindings_version": "23.1.1", "session_id": "b193bfa9-a5a9-44ae-bd88-ad94c471774b", "additional_info": [], "app_version": "1.0", "os_version": "16.0", "device_type": "arm64", "apple_pay_enabled": 1, "is_development": true, "client_filtered_payment_methods": "[\"sepa_debit\"]"]
Payment failed: noPaymentMethodTypesAvailable

**Comment by wooj-stripe** [comment]
Hi @irakliitavadze -

Thanks for reaching out!  I noticed that you are making a call to ` PaymentSheet(setupIntentClientSecret:` which tells me you are trying to use this with a setupIntent.  Unfortunately, `sepa_debit` is not a supported payment method for setupIntents, nor PI+SetupForFuture usage with payment sheet.  For now, you can only use sepa_debit with payment intents on payment sheet.

I'm curious to know more about your use case.  Are you needing support for setup intents, or payment intents + setup for future usage, or both?

References:
Corresponding [code](https://github.com/stripe/stripe-ios/blob/master/StripePaymentSheet/StripePaymentSheet/PaymentSheet/PaymentSheet%2BPaymentMethodAvailability.swift#L112)
Corresponding [test](https://github.com/stripe/stripe-ios/blob/master/Tests/Tests/PaymentSheet%2BPaymentMethodAvailabilityTest.swift#L173)

**Comment by irakliitavadze** [comment]
Hi @wooj-stripe -

Thank you for your reply!
Let me describe what we do and what we would like to implement: 
1. I receive secret token, customerId, publicKey from our backend.
 2. I need to show view where user types sepa details for a future payment.
 3. After successful input I should return paymentMethod ID to the backend.
4. Backend charges money when it will be need.

So it looks like I need SetupIntents API. But in this case I've to build own UI.
Maybe I missed something.

Could you please advise options to implant that flow on iOS.

Thanks in advance.

**Comment by wooj-stripe** [comment]
I see, unfortunately, `sepa_debit` for setupIntents and paymentIntents+SFU is not supported in payment sheet.  I will open an internal ticket for tracking to add this functionality, but can not provide any any timelines at the moment.  This means that if you want support for `sepa_debit` with setup intents, you will have to implement your own UI (and use the bindings, as you have demonstrated in your attached code which uses `STPPaymentMethodSEPADebitParams`).

As for why you are seeing `{"statusCode":400,"message":"Idempotency-Key not defined"}.` it sounds like you are confirming the setup intent on your backend (by calling `client?.confirmSetupIntent`)?  To help your debugging efforts, you might be able to confirm the setup intent on the client using `STPPaymentHandler`.

For example:
```
 let setupIntentParams = STPSetupIntentConfirmParams(clientSecret: secret)
 setupIntentParams.paymentMethodParams = paymentMethodParams
 STPPaymentHandler.shared().confirmSetupIntent(setupIntentParams, with: self) { status, intent, error in
   ....
  }
```

**Comment by yuki-stripe** [comment]
@irakliitavadze This is (finally) available in PaymentSheet 23.12.0 https://github.com/stripe/stripe-ios/releases/tag/23.12.0
